test/transparentproxy/install/install.go

```diff
@@ -0,0 +1,396 @@
+package install
+
+import (
+	"bytes"
+	"context"
+	std_errors "errors"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/exec"
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/kumahq/kuma/pkg/test/matchers"
+	. "github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/transparentproxy/utils"
+)
+
+var (
+	// Generic
+	cmdUserAdd = []string{"useradd", "-u", "5678", "kuma-dp"}
+	// Debian, Ubuntu
+	cmdAptUpdate          = []string{"apt-get", "update", "-y"}
+	cmdAptInstallIptables = []string{"apt-get", "install", "-y", "iptables"}
+	// RHEL, Amazon Linux
+	cmdYumUpdate          = []string{"yum", "update", "-y"}
+	cmdYumInstallIptables = []string{"yum", "install", "-y", "iptables"}
+	// On Amazon Linux, the `useradd` command is not available by default.
+	// The `shadow-utils` package, which provides user management tools,
+	// includes `useradd`.
+	cmdYumInstallShadowUtils = []string{"yum", "install", "-y", "shadow-utils"}
+	// Alpine
+	cmdAddUser        = []string{"adduser", "-u", "5678", "kuma-dp", "-D"}
+	cmdApkAddIptables = []string{"apk", "add", "iptables"}
+)
+
+type testCase struct {
+	name            string
+	image           string
+	postStart       [][]string
+	additionalFlags []string
+}
+
+func Install() {
+	DescribeTable(
+		"kumactl install transparent-proxy inside Docker container",
+		func(tc testCase) {
+			Expect(TProxyConfig.KumactlLinuxBin).NotTo(BeEmpty())
+
+			container, err := testcontainers.GenericContainer(
```

Comment:
It would be nice to wrap this into a builder as we do in other e2e tests so we can do something like:

```go
NewContainerSetup().
  WIthImage(tc.image).
  WithKumaExecutable(TProxyConfig.KumactlLinuxBin).
  WithPostStartHook(tc.postStart).
  WaitingForCmd([]string{"kumactl", "version"})
  Start()
```

Reply:
Addressed